dhcpv4/ztpv4/parse_circuitid.go

```diff
@@ -0,0 +1,97 @@
+package ztpv4
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/insomniacslk/dhcp/dhcpv4"
+)
+
+// CircuitID represents the structure of network vendor interface formats
+type CircuitID struct {
+	Slot		string
+	Module	string
+	Port		string
+	SubPort string
+	Vlan		string
+}
+
+var circuitRegexs = []*regexp.Regexp{
+	// Juniper QFX et-0/0/0:0.0
+	regexp.MustCompile(".*(?P<slot>[0-9]+)/(?P<mod>[0-9]+)/(?P<port>[0-9]+):(?P<subport>[0-9]+).*$"),
+	// Juniper PTX et-0/0/0.0
+	regexp.MustCompile(".*(?P<slot>[0-9]+)/(?P<mod>[0-9]+)/(?P<port>[0-9]+).*$"),
+	// Arista Ethernet3/17/1
+	// Juniper QFX et-1/0/61
+	regexp.MustCompile(".*(?P<slot>[0-9]+)/(?P<mod>[0-9]+)/(?P<port>[0-9]+)$"),
+	// Arista Ethernet14:Vlan2001
+	// Arista Ethernet10:2020
+	regexp.MustCompile(".*Ethernet(?P<port>[0-9]+):(?P<vlan>.*)$"),
+	// Cisco Gi1/10:2020
+	regexp.MustCompile(".*(?P<slot>[0-9]+)/(?P<port>[0-9]+):(?P<vlan>.*)$"),
+	// Nexus Ethernet1/3
+	regexp.MustCompile(".*(?P<slot>[0-9]+)/(?P<port>[0-9]+)$"),
+	// Juniper bundle interface ae52.0
+	regexp.MustCompile("^ae(?P<port>[0-9]+).(?P<subport>[0-9])$"),
+}
+
+// ParseCircuitId will parse dhcpv4 packet and return CircuitID info
+func ParseCircuitId(packet *dhcpv4.DHCPv4) (*CircuitID, error) {
```

Comment:
Love it, but wondering why this is in the ztpv4 module?

I'd have expected a GetCircuitID method on `dhcpv4.RelayOptions`:

https://github.com/insomniacslk/dhcp/blob/master/dhcpv4/option_relay_agent_information.go

Reply:
An example for how custom sub-options are implemented in BSDP: https://github.com/insomniacslk/dhcp/blob/master/dhcpv4/bsdp/option_vendor_specific_information.go

Reply:
>  why this is in the ztpv4 module?

This is not following an RFC. It is the parsing of relay agent information sub options, which according to the RFC are opaque values. It is dependant on each vendor. Same as with the `ztpv6` I think it is better to have it on a separate directory.

Reply:
I see, makes sense.

Reply:
@akshaynawale, do we actually need to return the whole circuit id struct or just the port id?

Otherwise, I would change this to `func ParseCircuitId(packet *dhcpv4.DHCPv4) (int, error)`

Reply:
I guess it's nice to have the whole structure and use only what you need of it (and the behaviour matches the function name, it should be changed otherwise)